src/user.rs

```diff
@@ -325,18 +341,45 @@ impl User<Synching> {
                 )))
                 .await;
                 self.state.rooms.lock().await.push(response.room_id.clone());
+                self.state
+                    .available_room_ids
+                    .push(response.room_id.to_string());
             }
             Err(e) => {
                 self.send(Event::Error((UserRequest::JoinRoom, e))).await;
             }
         }
     }
 
+    pub async fn add_friendship(&mut self, friendship: &Friendship) {
+        let client = self.client.lock().await.rooms();
+        if let Some(room) = client.iter().find(|room| {
+            if let Some(alias) = room.canonical_alias() {
+                return alias.alias().eq_ignore_ascii_case(&friendship.local_part);
+            }
+
+            false
+        }) {
+            self.state
+                .available_room_ids
+                .push(room.room_id().to_string());
+        }
+    }
+
     pub async fn act(&mut self) {
         let client = self.client.lock().await;
-        let rooms = self.state.rooms.lock().await;
+        let _rooms = self.state.rooms.lock().await;
+        let rooms = _rooms
```

Comment:
nit: You can use shadowing here and also avoid using `_` for prefixing variables that you are not ignoring.
```suggestion
        let rooms = self.state.rooms.lock().await;
        let rooms = rooms
```